src/components/LayerCreator.svelte

```diff
@@ -1,14 +1,32 @@
 <script context="module" lang="ts">
-	import GeoJsonLayer from './GeoJsonLayer.svelte';
+    import GeoJsonLayer from './GeoJsonLayer.svelte';
+    import GeotiffLayer from './GeotiffLayer.svelte';
     import LayerByIncome from './LayerByIncome.svelte';
     import type { Layer } from '../config';
 
-    export function CreateLayer(layerConfig: Layer) {
-        if (layerConfig.name === 'New York City Income') {
-            return {component: LayerByIncome, ...layerConfig};
+    export async function CreateLayer(layerConfig: Layer) {
+        let component;
+        /*
+        // FIXME add this code back in when we get the rollup config to work properly.
+        // As of now, the rollup makes one big bundle, which means we cannot dynamically
+        // load these components. Leaving it here as a reminder.
+        if (layerConfig.component !== undefined) {
+            // TODO is there any sort of security issue here?
+            component = (await import(`./${layerConfig.component}.svelte`)).default;
         }
+        */
 
-        return {component: GeoJsonLayer, ...layerConfig};
+        switch (layerConfig.type) {
+            case 'geojson':
+                component = GeoJsonLayer;
+                break;
+            case 'raster':
+                component = GeotiffLayer;
+                break;
+            default:
+                component = undefined;
+        }
+        return {component, ...layerConfig};
```

Comment:
I'd prefer not to use objects with anonymous types. For this particular case, it looks like `component` can just be returned because the contents of `layerConfig` aren't used. It would also be useful to log a warning when the layer type can't be found.

Reply:
We use the `layerConfig` to pass it along for all the props inside the actual layer. I have adjusted this to use more static typing.